docs/service_x509_credentials_aws.md

```diff
@@ -0,0 +1,346 @@
+
+It is assumed that at this point you have already configured the first
+EC2 Instance Profile IAM Role that your EC2 instance will be launched
+with. In order to access your EC2 instances through the Bastion daemon,
+your EC2 instance profile role must be named as `<domain>.<service>-service`.
+The bastion service uses the profile arn to determine the EC2 instance
+service name so it can carry out an authorization check to see if the user
+is authorized to ssh login to the specified instance or not. If the property
+is unable to name their instance profile name as `<domain>.<service>-service`
+then check out the [Non Standard Instance Profile Name](#non-standard-instance-profile-name)
+section for additional requirements.
+
+The second Athenz Service Identity IAM Assume Role must be created and
+must have the `<domain>.<service>` name. This role will not have any
+permissions but instead will have a trust relationship with your EC2
+instance role such that your EC2 instance role can assume this role.
+
+In the AWS Console, select `IAM` from the Services drop down and then
+click on the `Roles` link in the left sidebar. Choose the `Create Role`
+button. Under the `AWS Service` type, select `EC2`, then `EC2` again
+for the use case and finally choose `Next: Permissions` button in the
+bottom right corner.
+
+![IAM Role Setup_1](images/iam-role-setup-1.png)
+
+In the `Attach permissions policy` screen do not choose any permissions
+and just click on the `Next: Review` button in the bottom right corner
+to continue. Specify the `Role name` in the `<domain>.<service>` format
+and choose `Create Role` to complete the process.
+
+In the Roles list view, choose the role just created and choose the
+`Trust Relationships` tab.
+
+![IAM Role Setup_2](images/iam-role-setup-2.png)
+
+Click on `Edit trust relationship` button and append a block containing
+the following policy to the `Statement` block (Replace the `<account-id>`
+and `<ec2-instance-role>` values with their corresponding values for
+your environment:
+
+```
+ {
+   "Effect": "Allow",
+   "Principal": {
+     "AWS": "arn:aws:iam::<account-id>:role/<ec2-instance-role>"
+   },
+   "Action": "sts:AssumeRole"
+ }
+```
+
+Once correctly updated, your EC2 instance role must appear in the `Trusted entities` table:
+
+![IAM Role Setup_3](images/iam-role-setup-3.png)
+
+### Setup for Services with Region Specific Instance Profile Roles
+
+The IAM Role setup is identical when you have the same service
+being deployed in multiple regions with role names that
+are specific to each region. Since IAM roles are global, you only need
+to create a single Athenz IAM Role in the `<domain>.<service>` format.
+Then, when you're setting up your trust relationship, you'll authorize
+all your region specific profile roles to have the capability to
+assume the Athenz `<domain>.<service>` role.
+
+## Installing SIA
+
+The AWS SIA source is part of the Athenz project and can be found in:
```

Comment:
This area should be expanded as I'm not sure on the instructions for installing sia.
Also, not sure how it works with OpenStack as hca code isn't part of open-source.
Here is the original text:
 ## Base AMI Image
 Build your AMI image based on the foundation image provided by Platforms
 Organization that includes the Athenz SIA (Service Identity Agent)
 version 2.x. Based on the configuration file provided, this service
 will automatically contact Athenz ZTS service running in AWS, validate
 your instance identity and retrieve TLS Certificate for your instance.

 If you are using private foundation image then check out
 [Installing SIA without Foundation Image](#installing-sia-without-foundation-image)
 as how you can get Athenz SIA.